pkg/datapath/loader/loader.go

```
@@ -695,5 +697,22 @@ func (l *loader) HostDatapathInitialized() <-chan struct{} {
 }
 
 func (l *loader) WriteEndpointConfig(w io.Writer, e datapath.EndpointConfiguration) error {
-	return l.configWriter.WriteEndpointConfig(w, l.nodeConfig.Load(), e)
+	return l.configWriter.WriteEndpointConfig(w, l.getNodeConfig(), e)
+}
+
+func (l *loader) getNodeConfig() *datapath.LocalNodeConfiguration {
+	const retryInterval = 100 * time.Millisecond
+	for {
```

should we fail loudly after some (large) amount of time? say, 5 minutes?